pulp_python/app/tasks/repair.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import uuid
+from gettext import gettext as _
+
+from django.db.models.query import QuerySet
+from pulpcore.plugin.models import ProgressReport
+from pulpcore.plugin.util import get_domain
+
+from pulp_python.app.models import PythonPackageContent, PythonRepository
+from pulp_python.app.utils import artifact_to_python_content_data
+
+log = logging.getLogger(__name__)
+
+
+def repair(repository_pk: uuid.UUID) -> None:
+    """
+    Repairs metadata of all packages for the specified repository.
+
+    Args:
+        repository_pk (uuid.UUID): The primary key of the repository to repair.
+
+    Returns:
+        None
+    """
+    repository = PythonRepository.objects.get(pk=repository_pk)
+
+    log.info(
+        _(
+            "Repairing packages' metadata for the latest version of repository {}."
+        ).format(repository.name)
+    )
+    content_set = repository.latest_version().content.values_list("pk", flat=True)
+    content = PythonPackageContent.objects.filter(pk__in=content_set)
+
+    num_repaired = repair_metadata(content)
+    log.info(_("{} packages' metadata repaired.").format(num_repaired))
+
+
+def repair_metadata(content: QuerySet[PythonPackageContent]) -> int:
+    """
+    Repairs metadata for a queryset of PythonPackageContent objects
+    and updates the progress report.
+
+    Args:
+        content (QuerySet[PythonPackageContent]): The queryset of items to repair.
+
+    Returns:
+        int: The number of packages that were repaired.
+    """
+    # TODO: Add on_demand content repair
+    immediate_content = content.filter(contentartifact__artifact__isnull=False)
+    domain = get_domain()
+
+    batch = []
+    set_of_update_fields = set()
+    total_repaired = 0
+
+    progress_report = ProgressReport(
+        message="Repairing packages' metadata",
+        code="repair.metadata",
+        total=len(immediate_content),
```

Comment:
When you call `len` on the queryset you actually end up evaluating the whole queryset.
```suggestion
        total=immediate_content.count(),
```